apiserver/pkg/manager/codeflare_resource_manager.go

```diff
@@ -0,0 +1,129 @@
+type CodeFlareResourceManager struct {
+	DefaultResourceManager
+	mcadClient client.MCADClientInterface
+	scheme     *runtime.Scheme
+	converter  codeflare.AppWrapperConverter
+}
+
+// NewDefaultResourceManager provides a default implementation of the ResourceManager interface.
+func NewCodeFlareResourceManager(clientManager ClientManagerInterface, mcadClient client.MCADClientInterface) (*CodeFlareResourceManager, error) {
+	scheme := runtime.NewScheme()
+	err := rayv1api.AddToScheme(scheme)
+	if err != nil {
+		return nil, fmt.Errorf("error encountered while adding RayAPI to the scheme: %w", err)
+	}
+
+	err = mcadApi.AddToScheme(scheme)
+	if err != nil {
+		return nil, fmt.Errorf("error encountered while adding MCAD API to the scheme: %w", err)
+	}
+
+	awc := codeflare.NewAppWrapperConverter(scheme)
+	return &CodeFlareResourceManager{
+		DefaultResourceManager: *NewDefaultResourceManager(clientManager),
+		mcadClient:             mcadClient,
+		scheme:                 scheme,
+		converter:              awc,
+	}, nil
+}
+
+func (crm *CodeFlareResourceManager) CreateCluster(ctx context.Context, apiCluster *api.Cluster) (*rayv1api.RayCluster, error) {
+	namespace := apiCluster.Namespace
+	rayCluster, err := crm.DefaultResourceManager.createRayCluster(ctx, apiCluster, map[string]string{
+		"ray.io/codeflare-appwrapper-name": apiCluster.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create ray cluster CRD from api: %w", err)
+	}
+	clusterAppWrapper, err := crm.converter.AppWrapperFromRayCluster(rayCluster)
+	if err != nil {
+		return nil, fmt.Errorf("error creating appwrapper for ray cluster '%s/%s': %w", namespace, apiCluster.Name, err)
+	}
+	_, err = crm.mcadClient.MCADClient(namespace).Create(ctx, clusterAppWrapper, v1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cluster app wrapper: %w", err)
+	}
+	return rayCluster, nil
+}
+
+func (crm *CodeFlareResourceManager) GetCluster(ctx context.Context, clusterName string, namespace string) (*rayv1api.RayCluster, error) {
+	clusterAppWrapper, err := crm.mcadClient.MCADClient(namespace).Get(ctx, clusterName, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, util.NewNotFoundError(err, "Cluster %s not found", clusterName)
+		}
+		return nil, err
+	}
+	// The appwrapper is in a running state, therefore there should be a running cluster
+	// and is safe to get the state of the cluster from the kuberay api object
+	if clusterAppWrapper.Status.QueueJobState == mcadApi.AppWrapperCondDispatched &&
+		clusterAppWrapper.Status.State == mcadApi.AppWrapperStateActive {
+		return crm.DefaultResourceManager.GetCluster(ctx, clusterName, namespace)
+	}
+	return crm.converter.RayClusterFromAppWrapper(clusterAppWrapper)
+}
+
+func (crm *CodeFlareResourceManager) ListClusters(ctx context.Context, namespace string) ([]*rayv1api.RayCluster, error) {
+	return crm.DefaultResourceManager.ListClusters(ctx, namespace)
+}
```

Comment:
I am not convinced, this is correct, can we try to read clusters from MCAD? In your implementation, list will not return any of the clusters waiting for submission